modules/account-lib/src/coin/sol/instructionBuilders.ts

```diff
@@ -0,0 +1,78 @@
+import { PublicKey, SystemProgram, TransactionInstruction } from '@solana/web3.js';
+import assert from 'assert';
+import BigNumber from 'bignumber.js';
+import { InstructionBuilderTypes, MEMO_PROGRAM_PK } from './constants';
+import { InstructionParams, Memo, Nonce, Transfer, WalletInit } from './iface';
+
+export function instructionBuilder(instructionToBuild: InstructionParams): TransactionInstruction[] {
+  switch (instructionToBuild.type) {
+    case InstructionBuilderTypes.NonceAdvance:
+      return advanceNonceBuilder(instructionToBuild);
+    case InstructionBuilderTypes.Memo:
+      return memoBuilder(instructionToBuild);
+    case InstructionBuilderTypes.Transfer:
+      return transferBuilder(instructionToBuild);
+    case InstructionBuilderTypes.CreateNonceAccount:
+      return createNonceAccountBuilder(instructionToBuild);
+    default:
+      throw new Error(`Invalid instruction type or not supported`);
+  }
+}
+
+function advanceNonceBuilder(data: Nonce) {
```

Comment:
I don't know if the proper name for this function is good to call it builder.
I think just change for advanceNonceInstruction, because is not a builder pattern implemented here.

Reply:
Same for the other instructions.

Reply:
Renamed